src/BenchmarkDotNet/Running/BenchmarkRunnerClean.cs

```diff
@@ -531,14 +532,27 @@ private static BenchmarkReport RunCore(BenchmarkCase benchmarkCase, BenchmarkId
         private static void LogTotalTime(ILogger logger, TimeSpan time, int executedBenchmarksCount, string message = "Total time")
             => logger.WriteLineStatistic($"{message}: {time.ToFormattedTotalTime(DefaultCultureInfo.Instance)}, executed benchmarks: {executedBenchmarksCount}");
 
-        private static BenchmarkRunInfo[] GetSupportedBenchmarks(BenchmarkRunInfo[] benchmarkRunInfos, ILogger logger, IResolver resolver)
-            => benchmarkRunInfos.Select(info => new BenchmarkRunInfo(
-                    info.BenchmarksCases.Where(benchmark => benchmark.GetToolchain().IsSupported(benchmark, logger, resolver)).ToArray(),
+        private static (BenchmarkRunInfo[], List<ValidationError>) GetSupportedBenchmarks(BenchmarkRunInfo[] benchmarkRunInfos, ILogger logger, IResolver resolver)
+        {
+            List<ValidationError> validationErrors = new ();
+
+            var benchmarksRunInfo = benchmarkRunInfos.Select(info => new BenchmarkRunInfo(
+                    info.BenchmarksCases.Where(benchmark =>
+                    {
+                        var benchmarkValidationErrors = benchmark.GetToolchain().Validate(benchmark, resolver);
+
+                        validationErrors.AddRange(benchmarkValidationErrors);
+
+                        return !benchmarkValidationErrors.Any();
```

Comment:
"the more nested the type, the shorter its name" 

Bug: it will execute `Validation` twice.


```suggestion
                        var errors = benchmark.GetToolchain().Validate(benchmark, resolver).ToArray();
                        validationErrors.AddRange(errors);
                        return !errors.Any();
```